pkg/util/write.go

```diff
@@ -0,0 +1,28 @@
+package util
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+)
+
+func WriteIfChanged(filename string, data []byte, perm os.FileMode) (bool, error) {
```

Comment:
Kind of weird semantics. What if I want to keep the `perm` I had previously, but don't know it?